lib/super_digit.rb

```diff
@@ -1,15 +1,29 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity O(n) because I am using a loop ?
+# Space Complexity 0(n) becuase I am using recursion?
 def super_digit(n)
-   
+  string_n = n.to_s
+  return n if string_n.size == 1
+
+  i = 1
+  sum = string_n[0].to_i
+  while i < string_n.size
+    sum += string_n[i].to_i
+    i += 1
+  end
+  return super_digit(sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(k+n) where n is the number of digits in n?
+# Space Complexity - O(n) because I am using recursion ?
+
 def refined_super_digit(n, k)
```

Comment:
Actually your time complexity is O(n * k), you can bring it down to O(n + k) if you do it with a dynamic programming approach.

```ruby
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
```

Reply:
How? I'm confused. I will hop in your office hours tomorrow to go over these comments.